### PR TITLE
Create internal subpackage and move non public facing classes within

### PR DESCRIPTION
**Pull request by rymorale**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by sbenedicadb — APPROVED** [review]
looks good.  just one question on what to do with `PropositionPayload` since it's exposed as a public property in `Message` class but is itself internal

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java [hunk 1]
```diff
@@ -24,14 +20,19 @@
 
 import com.adobe.marketing.mobile.AdobeCallback;
 import com.adobe.marketing.mobile.launch.rulesengine.RuleConsequence;
-import com.adobe.marketing.mobile.messaging.MessagingConstants.EventDataKeys.MobileParametersKeys;
+import com.adobe.marketing.mobile.messaging.internal.MessageRequiredFieldMissingException;
```
Comment (praveek) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java:23]: https://github.com/adobe/aepsdk-sample-app-android/blob/main/Sample-App/app/src/main/java/com/adobe/marketing/mobile/sampleapp/MainApp.java#L15

In 1.x Messaging APIs are exposed inside `com.adobe.marketing.mobile` package.

Comment (praveek) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java:23]: Is this a new API we expose in 2.0.0?

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java [hunk 2]
```diff
@@ -286,83 +306,83 @@ private MessageSettings messageSettingsFromMap(final Map<String, Object> rawSett
         MessageAnimation displayAnimation, dismissAnimation;
         Map<MessageGesture, String> gestureMap = new HashMap<>();
 
-        if (rawSettings.get(MobileParametersKeys.WIDTH) != null) {
-            width = (int) rawSettings.get(MobileParametersKeys.WIDTH);
+        if (rawSettings.get(WIDTH) != null) {
+            width = (int) rawSettings.get(WIDTH);
```
Comment (praveek) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java:310]: Can you use DataReader util for reading from map in a type safe way?

File: code/messaging/src/main/java/com/adobe/marketing/mobile/Message.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    /**
+     * Returns the {@link Message}'s id.
+     */
+    String getId();
+
+    /**
+     * Returns the {@link MessagingExtension} which created this {@link Message} object.
+     */
+    MessagingExtension getParent();
```
Comment (praveek) [inline code/messaging/src/main/java/com/adobe/marketing/mobile/Message.java:75]: Can you change it to Object instead?

**Review by praveek — APPROVED** [review]

File: code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java [hunk 4]
```diff
@@ -51,7 +71,7 @@ public class Message extends MessagingDelegate {
     // private properties
     private WebView webView;
     // package private
-    PropositionInfo propositionInfo; // contains XDM data necessary for tracking in-app interactions with Adobe Journey Optimizer
+    public PropositionInfo propositionInfo; // contains XDM data necessary for tracking in-app interactions with Adobe Journey Optimizer
```
[the remark this answers is not in the document]
Comment (rymorale) [reply code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/Message.java]: i've cleaned this up and removed public from any classes within the internal subpackage which didn't need to be.

**Pull request merged**